packages/react-router-dom/server.tsx

```diff
@@ -60,55 +63,40 @@ export function StaticRouter({
 }
 
 export interface DataStaticRouterProps {
-  dataRoutes: DataRouteObject[];
+  children?: React.ReactNode;
   context: StaticHandlerContext;
+  dataRoutes: DataRouteObject[];
 }
 
 /**
  * A Data Router that may not navigate to any other location. This is useful
  * on the server where there is no stateful UI.
  */
 export function DataStaticRouter({
-  dataRoutes,
+  children,
   context,
+  dataRoutes,
 }: DataStaticRouterProps) {
   invariant(
     dataRoutes && context,
     "You must provide `routes` and `context` to <DataStaticRouter>"
   );
 
-  let router = getStatelessRouter(dataRoutes, context);
+  let dataRouterContext = {
+    router: getStatelessRemixRouter(dataRoutes, context),
+    navigator: getStatelessNavigator(),
+    static: true,
+    basename: "/",
+  };
   return (
-    <DataRouterContext.Provider value={router}>
-      <DataRouterStateContext.Provider value={router.state}>
-        <Router
-          location={router.state.location}
-          navigationType={router.state.historyAction}
-          navigator={getStatelessNavigator()}
-          static={true}
-        >
-          <DataStaticRoutes
-            dataRoutes={router.routes}
-            location={router.state.location}
-          />
-        </Router>
+    <DataRouterContext.Provider value={dataRouterContext}>
+      <DataRouterStateContext.Provider value={dataRouterContext.router.state}>
+        {children ? <>{children}</> : <DataRouter />}
```

Comment:
We now support children here if you want to take control of rendering your own `DataRouter`:

```jsx
// Simple SSR use cases
<DataStaticRouter dataRoutes={dataRoutes} context={context} />

// Advanced (i.e., Remix) use cases.  RemixEntry renders some context + boundaries 
// and then it's own DataRouter
<DataStaticRouter dataRoutes={dataRoutes} context={context}>
  <RemixEntry context={entryContext} />
</DataStaticRouter>
```

Reply:
Removed